runtime/major_gc.c

```diff
@@ -554,10 +563,136 @@ static void mark_ephe_aux (struct mark_stack *stk, intnat *work,
   }
 }
 
+
+#define Pb_size 256
+#define Pb_min 64
+#define Pb_mask (Pb_size - 1)
+#define Queue_prefetch_distance 4
+
+static uintnat rotate1(uintnat x)
+{
+  return (x << ((sizeof x)*8 - 1)) | (x >> 1);
+}
+
+CAMLnoinline static intnat do_some_marking(intnat work)
+{
+  uintnat pb_enqueued = 0, pb_dequeued = 0;
+  int darkened_anything = 0;
+  value pb[Pb_size];
+  uintnat min_pb = Pb_min;
+  struct mark_stack stk = *Caml_state->mark_stack;
+
+  uintnat young_start = (uintnat)Caml_state->young_start;
+  uintnat half_young_len = ((uintnat)Caml_state->young_end - (uintnat)Caml_state->young_start) >> 1;
```

Comment:
Check-typo says this line is too long.
```suggestion
  uintnat half_young_len =
    ((uintnat)Caml_state->young_end - (uintnat)Caml_state->young_start) >> 1;
```